runtime/src/lib.rs

```diff
@@ -443,6 +443,26 @@ impl_runtime_apis! {
 					.collect(),
 			}
 		}
+
+		/// Returns all the current auctions without auction id,
+		///  or return a specific auction info with an auction id.
+		fn current_auctions(auction_id: Option<AuctionId>) -> Vec<AuctionInfo>{
+			let data = pallet_auction::Auctions::<Runtime>::iter()
+			.map(|(auction, auction_info)|{
+				AuctionInfo { auction_id: auction, nft_id: auction_info.nft_id, start: auction_info.start, reserve: auction_info.reserve, buy_now: auction_info.buy_now, expiry_block: auction_info.expiry_block, current_bid: auction_info.current_bid }
```

Comment:
you may have to manually format these.

Cargo fmt have difficulties formatting code within Macros